Emby.Server.Implementations/Library/DotIgnoreIgnoreRule.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.IO;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Resolvers;
+using MediaBrowser.Model.IO;
+
+namespace Emby.Server.Implementations.Library;
+
+/// <summary>
+/// Resolver rule class for ignoring files via .ignore.
+/// </summary>
+public class DotIgnoreIgnoreRule : IResolverIgnoreRule
+{
+    private static FileInfo? FindIgnoreFile(DirectoryInfo directory)
+    {
+        var ignoreFile = new FileInfo(Path.Join(directory.FullName, ".ignore"));
+        if (ignoreFile.Exists)
+        {
+            return ignoreFile;
+        }
+
+        var parentDir = directory.Parent;
+        if (parentDir == null || parentDir.FullName == directory.FullName)
+        {
+            return null;
+        }
+
+        return FindIgnoreFile(parentDir);
+    }
+
+    /// <inheritdoc />
+    public bool ShouldIgnore(FileSystemMetadata fileInfo, BaseItem? parent)
+    {
+        return IsIgnored(fileInfo, parent);
+    }
+
+    /// <summary>
+    /// Checks whether or not the file is ignored.
+    /// </summary>
+    /// <param name="fileInfo">The file information.</param>
+    /// <param name="parent">The parent BaseItem.</param>
+    /// <returns>True if the file should be ignored.</returns>
+    public static bool IsIgnored(FileSystemMetadata fileInfo, BaseItem? parent)
```

Comment:
I think this should be cached, right now it will attempt to locate and read the `.ignore` file for every single file

Reply:
Yeah I was thinking about this too. Though there wasn't a big impact on my scan time  with the PR merged.

Reply:
Probably fine to not cache for now but we may need to revisit based on user reports
The bigger thing would be ensuring that the cache is invalidated when library scan starts so we don't keep ignoring after the .ignore file is changed